### PR TITLE
KEYCLOAK-14952 - Unit test failure in keycloak-saml-core on Java 11

### PR DESCRIPTION
**Pull request by dteleguin**

### CODE REVIEW

**Review by pedroigor — DISMISSED** [review]
@hmlnarik These changes look OK to me although I'm not sure about the second one due to the fact that in theory responses will include the XML declaration and that might be a problem if people is expecting the response to not have one (e.g.: including in another document although that is unlikely because usually, you care about the SAML Assertion).

I also don't have a better solution.

File: saml-core/src/main/java/org/keycloak/saml/common/util/DocumentUtil.java [hunk 1]
```diff
@@ -304,6 +305,16 @@ public static InputStream getSourceAsStream(Source source) throws ConfigurationE
         Result streamResult = new StreamResult(baos);
         // Write the DOM document to the stream
         Transformer transformer = TransformerUtil.getTransformer();
+
+        if (DOMSource.class.isInstance(source)) {
+            Node node = ((DOMSource) source).getNode();
+            if (Document.class.isInstance(node)) {
+                String xmlEncoding = ((Document) node).getXmlEncoding();
+                if (xmlEncoding != null)
+                    transformer.setOutputProperty(OutputKeys.ENCODING, xmlEncoding);
```
Comment (hmlnarik) [inline saml-core/src/main/java/org/keycloak/saml/common/util/DocumentUtil.java:314]: I'd rather suggest keeping TransformerUtil as is and including the xml declaration if necessary due to encoding difference. This is few bytes difference yet that can be helpful in redirect binding.
```suggestion
                    transformer.setOutputProperty(OutputKeys.ENCODING, xmlEncoding);
                    transformer.setOutputProperty(OutputKeys.OMIT_XML_DECLARATION, "no");
```

**Review by hmlnarik — CHANGES_REQUESTED** [review]
Thank for the fix! Please see inline

File: saml-core/src/main/java/org/keycloak/saml/common/util/DocumentUtil.java (hunk 1 above)
Comment (dteleguin) [reply saml-core/src/main/java/org/keycloak/saml/common/util/DocumentUtil.java:314]: done

**Review by hmlnarik — APPROVED** [review]
Thank you, @dteleguin !

**Pull request merged**